### DISCUSSION TITLE
Manual probe UBL and Zstop switch

### DISCUSSION THREAD

**Discussion by melvinmicallef** [Jyers/Marlin#315] [Q&A] [answered]
Hi, could anyone let me know when using this firmware compiled with manual UBL, what shall i do with the physical end stop switch? Should i lower it down? If so, does it mean the the z height is dependent on the mesh only and if something goes wrong with the mesh (or it is not loaded), the nozzle will hit the glass? Thanks

**Comment by Jyers** [comment, marked as answer]
With any manual leveling version you still very much need your endstop. Unless you have a bed probe, your z endstop will always be required to home your printer, the mesh just corrects this home position as the printhead moves across the bed.

**Reply by melvinmicallef** [reply]
Thank you and sorry if this was a stupid question :P Anyways.. this firmware rocks!!

**Reply by Jyers** [reply]
No such thing :P Glad you're enjoying it!
